src/Microsoft.AspNet.Mvc.Core/ModelBindingHelper.cs

```diff
@@ -293,22 +293,19 @@ public static class ModelBindingHelper
                 ModelBinder = modelBinder,
                 ValidatorProvider = validatorProvider,
                 MetadataProvider = metadataProvider,
-                HttpContext = httpContext
-            };
-
-            var modelBindingContext = new ModelBindingContext
-            {
-                Model = model,
-                ModelMetadata = modelMetadata,
-                ModelName = prefix,
-                ModelState = modelState,
+                HttpContext = httpContext,
                 ValueProvider = valueProvider,
```

Comment:
No, nothing had to be changed in the tests. It's possible that we had a bug in the case where a model property needs to 'reset' the value provider.